cmd/gossamer/config.go

```diff
@@ -0,0 +1,232 @@
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ChainSafe/gossamer/dot"
+
+	log "github.com/ChainSafe/log15"
+	"github.com/urfave/cli"
+)
+
+// createDotConfig creates a new dot configuration from the provided flag values
+func createDotConfig(ctx *cli.Context) (cfg *dot.Config, err error) {
+	// check --node flag and load node configuration from defaults.go
+	if name := ctx.GlobalString(NodeFlag.Name); name != "" {
+		switch name {
+		case "gssmr":
+			log.Info("[cmd] Loading node implementation...", "name", name)
+			cfg = dot.GssmrConfig() // "gssmr" = dot.GssmrConfig()
+		case "ksmcc":
+			log.Info("[cmd] Loading node implementation...", "name", name)
+			cfg = dot.KsmccConfig() // "ksmcc" = dot.KsmccConfig()
+		default:
+			return nil, fmt.Errorf("unknown node implementation: %s", name)
+		}
+	}
+
+	// check --config flag and load toml configuration from config.toml
+	if config := ctx.GlobalString(ConfigFlag.Name); config != "" {
+		log.Info("[cmd] Loading toml configuration...", "config", config)
+		if cfg == nil {
+			cfg = &dot.Config{} // if configuration has not been set, create empty dot configuration
+		} else {
+			log.Warn(
+				"[cmd] Overwriting node implementation with toml configuration values",
+				"name", cfg.Global.Name,
+				"config", config,
+			)
+		}
+		err = dot.LoadConfig(cfg, config) // load toml configuration values into dot configuration
+		if err != nil {
+			log.Error("[cmd] Failed to load toml configuration", "error", err)
+			return nil, err
+		}
+	}
+
+	// if configuration has not been set, load "gssmr" node implemenetation from node/gssmr/defaults.go
+	if cfg == nil {
+		log.Info("[cmd] Loading node implementation...", "name", "gssmr")
+		cfg = dot.GssmrConfig()
+	}
+
+	// set dot configuration values
+	setDotGlobalConfig(ctx, &cfg.Global)
+	setDotAccountConfig(ctx, &cfg.Account)
+	setDotNetworkConfig(ctx, &cfg.Network)
+	setDotCoreConfig(ctx, &cfg.Core)
+	setDotRPCConfig(ctx, &cfg.RPC)
+
+	// return dot configuration ready for node initialization
+	return cfg, nil
+}
+
+// setDotGlobalConfig sets dot.GlobalConfig using flag values from the cli context
+func setDotGlobalConfig(ctx *cli.Context, cfg *dot.GlobalConfig) {
+	// check --config flag and update node confifuration
+	if config := ctx.GlobalString(ConfigFlag.Name); config != "" {
+		cfg.Config = config
+	}
+
+	// check --genesis flag and update node confifuration
+	if genesis := ctx.GlobalString(GenesisFlag.Name); genesis != "" {
```

Comment:
not really a global flag, we shouldn't be able to use this if the node is initialized, or unless we are doing `init`